### PR TITLE
Prevent mutating $type in the schema while processing requests

### PR DESCRIPTION
**Pull request by simhnna**
I'm relying on the wsdl for typechecking and requests mutate `$type`.

I don't really understand much about the code, but with my change `$type` is no longer changed and all tests still pass.

This behavior was introduced with an old [commit](https://github.com/vpulim/node-soap/commit/9b6ba1fa39a4c15abf71d108cc7d159d5a8e3645) and I would consider changing the schema during request processing to be a bad habit. Ideally the namespaces would be set initially during schema parsing and not touched later.

### CODE REVIEW

File: test/client-schema-does-not-change-on-request-test.js [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+function cbCaller(client, methodName, requestJSON, responseJSON, responseSoapHeaderJSON, options, attachmentParts, done){
+    client[methodName](requestJSON, function(err, json, body, soapHeader){
+      if(requestJSON){
+        if (err) {
+          assert.notEqual('undefined: undefined', err.message);
+          assert.deepEqual(err.root, responseJSON);
+        } else {
+          // assert.deepEqual(json, responseJSON);
+          assert.equal(JSON.stringify(typeof json === 'undefined' ? null : json), JSON.stringify(responseJSON));
```
Comment (w666) [inline test/client-schema-does-not-change-on-request-test.js:113]: I am wondering what was the reason to use stringify and equal instead of deepEqueal?

Comment (simhnna) [reply test/client-schema-does-not-change-on-request-test.js:113]: That's shamelessly copied from the request/response tests...

Comment (w666) [reply test/client-schema-does-not-change-on-request-test.js:113]: Does it work with the deepEqual?

Comment (simhnna) [reply test/client-schema-does-not-change-on-request-test.js]: done

**Pull request merged**